pkg/provisioner/controller.go

```diff
@@ -220,12 +221,18 @@ func (c *Controller) syncHandler(key string) error {
 func (c *Controller) handleProvisionClaim(key string, obc *v1alpha1.ObjectBucketClaim, class *storagev1.StorageClass) (err error) {
 
 	var (
-		ob        *v1alpha1.ObjectBucket
-		secret    *corev1.Secret
-		configMap *corev1.ConfigMap
+		ob               *v1alpha1.ObjectBucket
+		secret           *corev1.Secret
+		configMap        *corev1.ConfigMap
+		provisionerLabel = map[string]string{c.provisionerName: ""}
```

Comment:
I wouldn't use empty string value. 
That's too confusing for no good reason.
How about the value will be provided by the operator that initializes the library?
I think it makes sense that each operator will use its version as the label value so that we can identify which version of the operator created those?

Reply:
The provisioner already supplies their name in the `NewProvisione` lib call so perhaps this can be used as the value here?

Reply:
the name is useless - the version will help troubleshooting much more.

Reply:
Are these labels for trouble shooting or making cleanup simpler?  Version is probably useful but is not currently passed to the library.

Reply:
Both. So lets add another ctor param for label value.

Reply:
@jeffvance 
> The provisioner already supplies their name in the NewProvisioner lib call so perhaps this can be used as the value here?

see PR description of why we have to use it as the key and not the value

@guymguym 
> the name is useless - the version will help troubleshooting much more.

I don't really agree - obcs in particular don't identify their provisioner.  Labeling our resources allows for getting them by their provisioner, which I definitely useful.  e.g.
`kubectl get -l <provisionerName> obc,secret,cm --all-namespaces` or
`kubectl get ob -l <provisionerName>`

This works even if the value is empty.

> That's too confusing for no good reason.

I'm not following this, why would it be confusing?

Reply:
An empty string value is confusing. Certainly in go.

The name is only useless as a label value, since the key is already the provisioner name...

I thought what is the most helpful value, and the operator version is the best I came up with. Do you see a better option?